Src/FluentAssertions/CallerIdentification/QuotesParsingStrategy.cs

```diff
@@ -54,17 +53,7 @@ public void NotifyEndOfLineReached()
 
     private bool IsVerbatim(StringBuilder statement)
     {
-        return
-            statement.Length >= 1
-            &&
-                new[]
-                {
-                    "$@",
-                    "@$",
-                }
-                .Any(verbatimStringOpener =>
-                    previousChar == verbatimStringOpener[1]
-                    && statement[statement.Length - 1] == verbatimStringOpener[1]
-                    && statement[statement.Length - 2] == verbatimStringOpener[0]);
+        return (previousChar is '@' && statement.Length >= 2 && statement[^2] is '$' && statement[^1] is '@')
+            || (previousChar is '$' && statement.Length >= 2 && statement[^2] is '@' && statement[^1] is '$');
```

Comment:
|  Method | previousChar | statement |       Mean |     Error |    StdDev | Ratio | Allocated |
|-------- |------------- |---------- |-----------:|----------:|----------:|------:|----------:|
| develop |         null |        "" |  5.7070 ns | 0.0368 ns | 0.0344 ns |  1.00 |      32 B |
|      PR |         null |        "" |  1.0103 ns | 0.0094 ns | 0.0088 ns |  0.18 |         - |
|         |              |           |            |           |           |       |           |
| develop |         null |        $@ | 38.2595 ns | 0.0580 ns | 0.0484 ns |  1.00 |     168 B |
|      PR |         null |        $@ |  1.0502 ns | 0.0056 ns | 0.0053 ns |  0.03 |         - |
|         |              |           |            |           |           |       |           |
| develop |            $ |        "" |  5.4029 ns | 0.0085 ns | 0.0075 ns |  1.00 |      32 B |
|      PR |            $ |        "" |  1.1945 ns | 0.0082 ns | 0.0069 ns |  0.22 |         - |
|         |              |           |            |           |           |       |           |
| develop |            $ |        $@ | 42.4361 ns | 0.4080 ns | 0.3817 ns |  1.00 |     168 B |
|      PR |            $ |        $@ |  2.5188 ns | 0.0063 ns | 0.0059 ns |  0.06 |         - |
|         |              |           |            |           |           |       |           |
| develop |            @ |        "" |  5.4685 ns | 0.0397 ns | 0.0372 ns |  1.00 |      32 B |
|      PR |            @ |        "" |  1.2068 ns | 0.0064 ns | 0.0057 ns |  0.22 |         - |
|         |              |           |            |           |           |       |           |
| develop |            @ |        $@ | 38.2881 ns | 0.4913 ns | 0.4595 ns |  1.00 |     168 B |
|      PR |            @ |        $@ |  3.5240 ns | 0.0278 ns | 0.0247 ns |  0.09 |         - |

Reply:
Wow

Reply:
Why is it so much faster?

Reply:
[SharpLab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABADAAgwRgDoAZASwDsBHAbgFgAobPIgFRgQBc76HyBDALYwAzgAc+YGDgBiAGwCuMchwCCw4TCgdSEcsMIBhPrNmaAkgBMl2gGakwfbbu4MKHTf1l4ATDgCK8hDuwgAKfFDCFADmAMocUI4wUQCeDADeDDhZOKJQpABuiThgABbhAPw5sPk68sIGZVAu9NlVBUXAEBBeAPKyFgAUcXnkUQBC8qT9mjjCHIlCygCUmdkZLa3ZGADsq5ubcwvWJEpRHCU4AHwAvDj4e/vZAGRPD4+b5DAA7gDaALpvd5rQFAzYAIgAJBgMGCADQg0HZMHQiFwhHvAC+6MehBU5GSA3ymmAjlIAmG0R6oiUM2ul2xoNyMBqEDqDXCOGutyJUBJ2nJ8Up1M+UB++ABG0Rjxes3m7kWHB+h3lx2Ip3OODgdz+nO5xNJApGUSpNNF4oZQJlypgCqVcptqvVFy13h1XJwPL5ZIpoxNIp+WD+S24rSxkpwD1y7XcOE63RwADlvkNBaMJlMrFBZUdlg91u8djgBkyWWzGjhSMIcAByaHVnBW+0Kk6jDU3HC+Rs5xUAPVdFar1Yh9a7KuUPx74oHNbrK3D7zQaCLJdq9XLlZrw4bT2zY44LbOF3bnZ31ttfZ1G9rGBHp6b1gnU6vw+DDzDYaAA=)


![image](https://user-images.githubusercontent.com/919634/210330982-3ecfb78e-995c-40e4-bcc8-9b614a08fd84.png)

Reply:
It could even be a _tiny_ bit faster using `previousChar.GetValueOrDefault() is '@'`, as `previousChar is '@'` includes a redundant `.HasValue` check, but I found this code cleaner.

https://github.com/dotnet/roslyn/issues/60413 and https://github.com/dotnet/roslyn/issues/52629 track removing redundant `.HasValue` checks.

Reply:
This change also fix a (potential) bug, since the old code checked for `statement.Length >= 1` but indexed into `statement.Length - 2`.

Reply:
Also for inspiration, with C#11 this can utilize [list patterns](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/patterns#list-patterns) and be written as

```cs
return (previousChar is '@' && statement is [.., '$', '@'])
    || (previousChar is '$' && statement is [.., '@', '$']);
```

The only _microscopic_ downside is that as far as I can tell, it will always generate a null check on `statement` which reduces the CC from 100%.